app-games/src/main/java/com/aptoide/android/aptoidegames/gamegenie/presentation/GameGenieManager.kt

```diff
@@ -81,6 +106,29 @@ class GameGenieManager @Inject constructor(
     }
   }
 
+  suspend fun saveOrUpdateChatCompanion(packageName: String, chat: GameGenieChat) {
+    val oldChat = getGameCompanionChat(packageName)
+    val newChat = chat.toCompanionEntity(packageName)
+    if (oldChat != null) {
+      val lastMessage = oldChat.conversation.last()
+        .copy(
+          user = chat.conversation[chat.conversation.size - 2].user
+        )
+      val fixedOldChat = oldChat.conversation.dropLast(1) + lastMessage
+      gameGenieDatabase
+        .getGameCompanionDao()
```

Comment:
i would inject the interface of the dao directly to not have to go and get the dao everytime. You will never use the database directly, only the dao